src/aws-cpp-sdk-core/include/aws/core/client/ClientConfiguration.h

```diff
@@ -165,9 +197,10 @@ namespace Aws
              */
             unsigned long lowSpeedLimit = 1;
             /**
-             * Strategy to use in case of failed requests. Default is DefaultRetryStrategy (i.e. exponential backoff)
+             * Strategy to use in case of failed requests. Default is DefaultRetryStrategy (i.e. exponential backoff).
+             * Provide retry strategy here or via a factory method.
              */
-            std::shared_ptr<RetryStrategy> retryStrategy;
+            std::shared_ptr<RetryStrategy> retryStrategy = nullptr;
```

Comment:
should we make a `ClientConfiguration` constructor that takes a `ProviderFactories` instead of push the logic to the client?

Reply:
We can do it later.
Additionally, I see no much extra harm in default-initing to default factories and then allowing users to over-write.